hibernate-core/src/main/java/org/hibernate/annotations/SQLRestriction.java

```diff
@@ -62,6 +63,7 @@
  */
 @Target({TYPE, METHOD, FIELD})
 @Retention(RUNTIME)
+@Inherited
```

Comment:
I'm not yet convinced that we should support the requested feature.

But if we decided to, this isn't the right fix, since it implies inheritance even from things which _aren't_ `@MappedSuperclass`es.

Reply:
Fair enough. I will fix implementation.

If I do so what will be the process of approving this feature? What is your reservations against it?

Reply:
I found that similar annotation `SoftDelete` is supporting `MappedSuperclass`. 

Will this convince you to accept my improvement?

Reply:
> Will this convince you to accept my improvement?

I don't hate it. Let's see what others think.

Reply:
Well it certainly can't be merged in its current form because it breaks the overridability of @SQLRestriction, as I commented yesterday. 

But if what you're saying is that this functionality is really not very important to you after all, then yes, please just close the PR.